### PR TITLE
Apply automated clippy fixes

### PR DESCRIPTION
**Pull request by tversteeg**
This is done using `cargo clippy --fix -Z unstable-options` I will try to fix some of the warnings later.

### CODE REVIEW

File: src/sdf.rs [hunk 1]
```diff
@@ -24,7 +29,7 @@ pub fn march_circle<F: Fn(Real2)->Real>(start: Real2, end: Real2, radius: Real,
     let mut l = reali(0);
     let mut hit = false;
     let forward_step = reali(1);
-    let backward_step = Real::from(0.125);
```
Comment (koalefant) [inline src/sdf.rs:27]: Is this a manual change or automatic clippy change?
I intend to build this with Real being a type-def of a fixed point type, and then I will need similar conversions of constants...

Comment (tversteeg) [reply src/sdf.rs:27]: This is an automatic clippy change. If you implement the `From` trait those conversions will be done automatically right?

Comment (koalefant) [reply src/sdf.rs:27]: I thought one have to call .into() or Real::from() explicitly, but I may be wrong.

Comment (koalefant) [reply src/sdf.rs:27]: I will take this and just replace it with a function later so it does not mess with clippy.

**Pull request merged**